src/Microsoft.OData.Core/Batch/ODataBatchUtils.cs

```diff
@@ -110,12 +110,13 @@ internal static Uri CreateOperationRequestUri(Uri uri, Uri baseUri, IODataPayloa
         /// <returns>A new <see cref="ODataWriteStream"/> instance.</returns>
         internal static ODataWriteStream CreateBatchOperationWriteStream(
             Stream outputStream,
-            IODataStreamListener operationListener)
+            IODataStreamListener operationListener,
+            bool synchronous = true)
         {
             Debug.Assert(outputStream != null, "outputStream != null");
             Debug.Assert(operationListener != null, "operationListener != null");
 
-            return new ODataWriteStream(outputStream, operationListener);
+            return new ODataWriteStream(outputStream, operationListener, synchronous);

```

Comment:
Could you remind me what the `synchronous` argument does again? I remember it from a previous PR but don't remember what it was used for. I think it was used in the dispose methods, but I'm not quite sure.

Reply:
Reference to https://github.com/OData/odata.net/pull/2109, `ODataWriteStream` previously had no way of telling whether we're writing synchronously or asynchronously. We address that gap by introducing this argument such that when the `Dispose` method is invoked, either of `StreamDisposed` or `StreamDisposedAsync` is called depending on the value passed for the argument.
This approach is necessitated by the fact that we support `NET45` and `NETSTANDARD1_1` both of which do not support `IAsyncDisposable` that would make it feasible to use `DisposeAsync` in asynchronous scenarios.